### PR TITLE
net: BSD Sockets poll() basic implementation

### PR DESCRIPTION
**Pull request by pfalcon**
This implements https://jira.zephyrproject.org/browse/ZEP-2228

### CODE REVIEW

File: subsys/net/lib/sockets/sockets.c [hunk 1]
```diff
@@ -375,3 +378,75 @@ int zsock_fcntl(int sock, int cmd, int flags)
...
+
+	if (timeout < 0) {
+		timeout = K_FOREVER;
+	}
+
+	pev = poll_events;
+	for (pfd = fds, i = nfds; i--; pfd++) {
+
+		/* Per POSIX, negative fd's are just ignored */
+		if (pfd->fd < 0) {
```
Comment (jukkar) [inline subsys/net/lib/sockets/sockets.c]: As the fd is just a pointer converted to int, I suppose its value could be < 0

File: subsys/net/lib/sockets/Kconfig [hunk 2]
```diff
@@ -25,6 +25,13 @@ config NET_SOCKETS_POSIX_NAMES
 	attention, as in POSIX it closes any file descriptor, while with this
 	option enaled, it will still apply only to sockets.
 
+config NET_SOCKETS_POLL_NUM
```
Comment (jukkar) [inline subsys/net/lib/sockets/Kconfig]: Should we call this ...._MAX instead, feels more logical that way.

File: subsys/net/lib/sockets/sockets.c (hunk 1 above)
Comment (pfalcon) [inline subsys/net/lib/sockets/sockets.c]: It's the basic assumption of the current implementation that it won't be negative (i.e. that RAM lies in first half of the CPU address space), e.g. https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/sockets/sockets.c#L78 . This fact will be documented in the docs (poll() is the last feature socket feature to make it feature-complete, so I'll work on docs in the next patches). If we face a case where RAM lies in 2nd half of address space, we can look how to deal with it. But I don't think we'll face such a case soon, at least I believe all boards currently supported by Zephyr have RAM in the 1st half.

File: subsys/net/lib/sockets/Kconfig (hunk 2 above)
Comment (pfalcon) [inline subsys/net/lib/sockets/Kconfig]: Let me grep thru Kconfigs and see how it's usually called, somehow I thought the convention is *_NUM.

Comment (pfalcon) [inline subsys/net/lib/sockets/Kconfig]: Recently introduce SYN backlog uses SIZE at all: CONFIG_NET_TCP_BACKLOG_SIZE. Anyway, I see that "MAX" is used in config options frequently, changing to it.

**Comment by pfalcon** [comment]
Updated per comments above.

**Review by jukkar — APPROVED** [review]
LGTM

**Pull request merged**